service/src/test/java/bio/terra/workspace/service/workspace/AzureWorkspaceTest.java

```diff
@@ -60,4 +96,110 @@ void createGetDeleteAzureContext() {
     workspaceService.deleteAzureCloudContext(workspace, userRequest);
     assertTrue(azureCloudContextService.getAzureCloudContext(workspace.getWorkspaceId()).isEmpty());
   }
+
+  @Test
+  void cloneAzureWorkspaceWithContainer() throws InterruptedException {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    Mockito.when(mockSamService.getUserEmailFromSamAndRethrowOnInterrupt(any()))
+        .thenReturn(userRequest.getEmail());
+
+    SpendProfileId spendProfileId = new SpendProfileId(UUID.randomUUID().toString());
+
+    UUID sourceUUID = UUID.randomUUID();
+    Workspace sourceWorkspace =
+        Workspace.builder()
+            .workspaceId(sourceUUID)
+            .userFacingId("a" + sourceUUID)
+            .spendProfileId(spendProfileId)
+            .workspaceStage(WorkspaceStage.MC_WORKSPACE)
+            .createdByEmail(userRequest.getEmail())
+            .build();
+
+    workspaceService.createWorkspace(sourceWorkspace, null, null, userRequest);
+
+    String jobId = UUID.randomUUID().toString();
+
+    workspaceService.createAzureCloudContext(
+        sourceWorkspace, jobId, userRequest, "/fake/value", azureTestUtils.getAzureCloudContext());
+    jobService.waitForJob(jobId);
+
+    assertTrue(
+        azureCloudContextService
+            .getAzureCloudContext(sourceWorkspace.getWorkspaceId())
+            .isPresent());
+
+    UUID landingZoneId = UUID.fromString(landingZoneTestUtils.getDefaultLandingZoneId());
+    String storageAccountName = String.format("lzsharedstacc%s", TestUtils.getRandomString(6));
+
+    TestLandingZoneManager testLandingZoneManager =
+        new TestLandingZoneManager(
+            azureCloudContextService,
+            landingZoneDao,
+            workspaceDao,
+            crlService,
+            azureConfig,
+            sourceWorkspace.getWorkspaceId());
+
+    testLandingZoneManager.createLandingZoneWithSharedStorageAccount(
+        landingZoneId, sourceWorkspace.getWorkspaceId(), storageAccountName, "eastus");
+
+    TimeUnit.MINUTES.sleep(1);
```

Comment:
This  1 minute sleep scares me. Can you poll for the resource to be created?

Reply:
I pulled this from other tests that used a LZ storage account, so not sure -- I'll look into it

Reply:
Yikes, we can't sprinkle 1 minute sleeps throughout. Yes, please follow up with the author of the other tests.

Reply:
It would be good to fix the other tests also (perhaps a helper method?) to speed up all the tests.

Reply:
I added a util and found that the offending tests have been removed in the past couple of days so we're all set